io.cucumber.eclipse.python/src/io/cucumber/eclipse/python/validation/BehaveGlueJob.java

```diff
@@ -0,0 +1,160 @@
+		if (project == null) {
+			return Status.CANCEL_STATUS;
+		}
+
+		try {
+			// Get behave command from preferences
+			BehavePreferences preferences = BehavePreferences.of(resource);
+			String behaveCommand = preferences.behaveCommand();
+			
+			// Run behave --dry-run --format steps.usage --no-summary
+			String workingDir = project.getLocation().toOSString();
+			String featurePath = resource.getLocation().toOSString();
+
+			BehaveProcessLauncher launcher = new BehaveProcessLauncher()
+				.withCommand(behaveCommand)
+				.withFeaturePath(featurePath)
+				.withWorkingDirectory(workingDir)
+				.withDryRun(true)
+				.withFormat("steps.usage")
+				.withNoSummary(true);
+
+			Process process = launcher.launch();
+
+			// Parse the output
+			Map<Integer, StepMatch> stepMatchMap = new HashMap<>();
+			Map<Integer, String> unmatchedSteps = new HashMap<>();
+			String currentStepPattern = null;
+			String currentStepFile = null;
+			int currentStepLine = -1;
+
+			try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
+				String line;
+				while ((line = reader.readLine()) != null) {
+					if (monitor.isCanceled()) {
+						process.destroy();
+						return Status.CANCEL_STATUS;
+					}
+
+					// Check if this is a step definition line
+					Matcher defMatcher = STEP_DEF_PATTERN.matcher(line);
+					if (defMatcher.find()) {
+						currentStepPattern = defMatcher.group(1);
+						currentStepFile = defMatcher.group(2);
+						currentStepLine = Integer.parseInt(defMatcher.group(3));
+						continue;
+					}
+
+					// Check if this is a step usage line
+					Matcher usageMatcher = STEP_USAGE_PATTERN.matcher(line);
+					if (usageMatcher.find() && currentStepPattern != null) {
+						String stepText = usageMatcher.group(2);
+						String featureFile = usageMatcher.group(3);
+						int featureLine = Integer.parseInt(usageMatcher.group(4));
+
+						// Only record matches for the current feature file
+						if (featureFile.contains(resource.getName())) {
+							StepMatch match = new StepMatch(featureLine, stepText, currentStepFile, currentStepLine,
+									currentStepPattern);
+							stepMatchMap.put(featureLine, match);
+						}
+					}
+				}
+			}
+
+			// Wait for process to complete
+			int exitCode = process.waitFor();
+
+			// Find all steps in the feature and determine which ones are unmatched
+			// For now, we'll just use the steps we found
+			matchedSteps = new ArrayList<>(stepMatchMap.values());
+
+			// Create markers for unmatched steps
+			// Collect snippets for steps that don't have matches
+			Map<Integer, Collection<String>> snippets = new HashMap<>();
+
+			// For demonstration, we'll assume all steps in the feature that aren't in stepMatchMap are unmatched
+			// In a real implementation, we'd parse the feature document to find all steps
+			
+			// Update markers
+			MarkerFactory.missingSteps(resource, snippets, Activator.PLUGIN_ID, false);
```

Comment:
This needs to be reworked / implemented

1. MarkerFactory.missingSteps seems not appropriate for our use case as we have no snippets to give to the user, instead we should have a new MarkerFactory.missingSteps method that works like the old ones but always uses an empty snippet, and only accepts a list of line numbers to add the marker
2. collect from the gherking document all parsed steps and remove from that list based on the line numbers we detect
3. The result can then be used to reliable determine what steps are actually missing and add the markers with the new method